blockchain/network.py

```diff
@@ -473,14 +549,54 @@ def resolve_data(self, node, guids, phase):
             try:
                 verification = thrift_converter.convert_thrift_verification(verification)
                 verification_db.insert_verification(verification, verification['verification_id'])
+                # check if there are nodes further down the chain interested in this record
                 replicated_verifications = verification_db.get_all_replication(verification['block_id'], phase, verification['origin_id'])
                 for replicated_ver in replicated_verifications:
                     vr_transfers_db.insert_transfer(replicated_ver['origin_id'], replicated_ver['signature']['signatory'], verification['verification_id'])
+                # check if there are active subscriptions interested in this record
+                self.update_subscription_response(verification)
```

Comment:
Very cool idea!
I think that it should happen around line 479, when we get a response for both VR and TX.